backend/core/users/verification.py

```
@@ -0,0 +1,171 @@
+	if current_app.config.get("USE_MEMORY_VERIFICATION_STORE", False):
+		return _MemoryRedisAdapter()
+	
+	redis_url = current_app.config.get("REDIS_URL", "redis://localhost:6379/0")
+	try:
+		return Redis.from_url(redis_url, decode_responses=True)
+	except Exception as exc:
+		# Fail fast on Redis misconfiguration in production
+		raise RuntimeError(
+			f"Failed to connect to Redis at {redis_url}. "
+			"Verify REDIS_URL is correct and Redis is running."
+		) from exc
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)